content/cloud-docs/api-docs/providers.mdx

```diff
@@ -254,25 +312,819 @@ curl \
+
+| Parameter            | Description                                                                                                                                                                                              |
+| -------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `:organization_name` | The name of the organization to create a provider in. The organization must already exist, and the token authenticating the API request must belong to the "owners" team or a member of the "owners" team. |
+| `:namespace`         | The namespace of the provider for which the version is being created. For private providers this is the same as the `:organization_name` parameter                                                           |
+| `:name`              | The name of the provider for which the version is being created.                                                                                                                                           |
+| `:registry_name`     | Must be `private`.                                                                                                                                                                                       |
+
+Creates a new registry provider version. This endpoint only applies to private providers.
+
+| Status  | Response                                                   | Reason                                                         |
+| ------- | ---------------------------------------------------------- | -------------------------------------------------------------- |
+| [201][] | [JSON API document][] (`type: "registry-provider-versions"`) | Successfully published provider version                      |
+| [422][] | [JSON API error object][]                                  | Malformed request body (missing attributes, wrong types, etc.) |
+| [403][] | [JSON API error object][]                                  | Forbidden - not available for public providers                   |
+| [404][] | [JSON API error object][]                                  | User not authorized                                            |
+
+### Request Body
+
+This POST endpoint requires a JSON object with the following properties as a request payload.
+
+Properties without a default value are required.
+
+| Key path                  | Type   | Default | Description                             |
+| ------------------------- | ------ | ------- | -------------------------------------   |
+| `data.type`               | string |         | Must be `"registry-provider-versions"`. |
+| `data.attributes.version` | string |         | A valid semver version string.          |
+| `data.attributes.key_id` | string |          | A valid gpg-key string.                 |
```

Comment:
This endpoint also takes a protocols field that's an array of one/all of the following values ["4.0","5.0","6.0"]

Reply:
Is it better to have nil or [] to be the default for the protocols field?